src/sassAutocomplete.ts

```diff
@@ -153,37 +143,86 @@ export function getValues(cssSchema, currentWord: string): CompletionItem[] {
     return completionItem;
   });
 }
+/**
+ * checks if currentWord last char is a number?
+ * @param {String} currentWord
+ * @return {CompletionItem}
+ */
+export function isNumber(currentWord: string): boolean {
+  const reg = /[0-9]$/;
+  return reg.test(currentWord);
+}
+/**
+ * gets variable completions.
+ * @param text
+ */
+export function getVariables(text: string) {
+  const regex = /\${1}\S*:/g;
+  let m;
+  const variables = [];
+
+  while ((m = regex.exec(text)) !== null) {
+    if (m.index === regex.lastIndex) {
+      regex.lastIndex++;
+    }
+    m.forEach((match: string) => {
+      const rep = match.replace(':', '');
+      const completionItem = new CompletionItem(rep.replace('$', ''));
+      completionItem.insertText = rep;
+      completionItem.detail = 'Sass Variable.';
+
+      completionItem.kind = CompletionItemKind.Variable;
+      variables.push(completionItem);
+    });
+  }
+  return variables;
+}
+/**
+ * gets unit completions.
+ * @param currentword
+ */
+export function getUnits(currentword: string) {
+  const units = [];
+
+  sassSchemaUnits.forEach(item => {
+    const lastWord = currentword.split(' ');
+    const rep = lastWord[lastWord.length - 1];
+    const completionItem = new CompletionItem(rep + item.name);
+    completionItem.insertText = new SnippetString(rep + item.body);
+    completionItem.detail = item.description;
+    completionItem.kind = CompletionItemKind.Unit;
+    units.push(completionItem);
+  });
+  return units;
+}
 
 class SassCompletion implements CompletionItemProvider {
-  provideCompletionItems(
-    document: TextDocument,
-    position: Position,
-    token: CancellationToken
-  ): CompletionItem[] {
+  provideCompletionItems(document: TextDocument, position: Position, token: CancellationToken): CompletionItem[] {
     const start = new Position(position.line, 0);
     const range = new Range(start, position);
     const currentWord = document.getText(range).trim();
+    const currentWordUT = document.getText(range);
     const text = document.getText();
     const value = isValue(cssSchema, currentWord);
     const config = workspace.getConfiguration('sass-indented');
 
     let atRules = [],
+      Units = [],
       properties = [],
-      values = [];
+      values = [],
+      variables = [];
 
+    if (isNumber(currentWordUT)) {
+      Units = getUnits(currentWord);
+    }
     if (value) {
       values = getValues(cssSchema, currentWord);
+      variables = getVariables(text);
```

Comment:
I can see variables being added to the completions list however the completion prompt is not displaying. This could be a settings issue on my end. 

As mentioned in the previous PR comments though I don't really see a huge value in being able to complete variables within the same file. Personally the majority of projects I have worked on (and as can be seen in most libraries) have had a single file for configuring variables to be used across the other project files. The benefit that jumps to mind is in a large config file like bootstraps where variables are being used to define others. I appreciate this portion wasn't part of your initial scope.